src/physics/convection.jl

```diff
@@ -168,37 +122,46 @@ function convection!(
         # precipitation is zero.
 
         for k in level_zero_buoyancy:nlayers
-            Qref -= humid_ref_profile[k]*Δσ[k]  # eq (11) but in σ coordinates
+            Qref -= humid_ref_profile[ij, k]*Δσ[k]  # eq (11) but in σ coordinates
         end
         fq = 1 - Pq/Qref                    # = 1 - Δq/Qref in eq (12) but we reuse Pq
 
         ΔT = PT/Δσ_lzb                      # equation (14), reuse PT and in σ coordinates
         for k in level_zero_buoyancy:nlayers
-            humid_ref_profile[k] *= fq      # update humidity profile, eq (13)
-            temp_ref_profile[k] -= ΔT       # update temperature profile, eq (15)
+            humid_ref_profile[ij, k] *= fq      # update humidity profile, eq (13)
+            temp_ref_profile[ij, k] -= ΔT       # update temperature profile, eq (15)
         end
     end
 
 	(; gravity) = planet
 	(; water_density) = atmosphere
 	(; Δt_sec) = time_stepping
 
+    # Initialize rain accumulation for this grid point
+    rain_convection = zero(NF)
+
     # GET TENDENCIES FROM ADJUSTED PROFILES
     for k in level_zero_buoyancy:nlayers
-        temp_tend[k] -= (temp[k] - temp_ref_profile[k]) / SBM.time_scale.value
-        δq = (humid[k] - humid_ref_profile[k]) / SBM.time_scale.value
-        humid_tend[k] -= δq
+        temp_tend[ij, k] -= (temp[ij, k] - temp_ref_profile[ij,k]) / SBM.time_scale.value
+        δq = (humid[ij, k] - humid_ref_profile[ij,k]) / SBM.time_scale.value
+        humid_tend[ij, k] -= δq
 
         # convective precipitation (rain), integrate dq\dt [(kg/kg)/s] vertically
         rain = max(δq * Δσ[k], zero(δq))        # only integrate excess humidity for precip (no reevaporation)
-        column.rain_convection += rain          # integrate vertically, Formula 25, unit [m]
+        rain_convection += rain                 # integrate vertically, Formula 25, unit [m]
     end
  
-    pₛΔt_gρ = (pₛ * Δt_sec / gravity / water_density) * deep_convection # enfore no precip for shallow conv
-	column.rain_convection *= pₛΔt_gρ                                   # convert to [m] of rain during Δt
-    column.rain_rate_convection = column.rain_convection / Δt_sec       # rate: convert to [m/s] of rain
+    pₛΔt_gρ = (pₛ * Δt_sec / gravity / water_density) * deep_convection # enforce no precip for shallow conv
+	rain_convection *= pₛΔt_gρ                                           # convert to [m] of rain during Δt
+    
+    # Store precipitation in diagnostic arrays
+    diagn.physics.rain_convection[ij] += rain_convection
+
+    # TODO: double check, because preivously this was rain_rate_convection, which didn't exist
+    diagn.physics.total_precipitation_rate[ij] = rain_convection / Δt_sec    # rate: convert to [m/s] of rain
```

Comment:
Just commenting here, to point out that I changed this here just so that I don't get any errors. It still needs to be corrected to the correct rate variable. 

But maybe the rain rates are also one thing were we can actually also accumulate all in one rate instead of separate ones to save some memory.